acceptance/lib/puppet/acceptance/common_utils.rb

```diff
@@ -6,5 +6,111 @@ def ruby_command(host)
       end
       module_function :ruby_command
     end
-  end
-end
+
+    module CAUtils
+      def initialize_ssl
+        hostname = on(master, 'facter hostname').stdout.strip
+        fqdn = on(master, 'facter fqdn').stdout.strip
+
+        if master.use_service_scripts?
+          step "Ensure puppet is stopped"
+          # Passenger, in particular, must be shutdown for the cert setup steps to work,
+          # but any running puppet master will interfere with webrick starting up and
+          # potentially ignore the puppet.conf changes.
+          on(master, puppet('resource', 'service', master['puppetservice'], "ensure=stopped"))
```

Comment:
Since you're always running against `puppetserver` you can change `master['puppetservice']` to just be `puppetserver`. Puppet had to account that the service name was sometimes `puppet` (webrick), `puppetserver` (JVM puppetserver), `apache2` or `httpd` (passenger-based puppet master).